src/home/rendering/render-github-login-button.ts

```diff
@@ -19,16 +19,48 @@ export async function checkSupabaseSession() {
 }
 
 async function gitHubLoginButtonHandler() {
-  const { error } = await supabase.auth.signInWithOAuth({
-    provider: "github",
-    options: {
-      scopes: "repo",
-    },
-  });
-  if (error) {
-    console.error("Error logging in:", error);
+  try {
+    const { user, error } = await supabase.auth.signInWithOAuth({
+      provider: "github",
+      options: {
+        scopes: "public_repo", // Default scope for newcomers
+      },
+    });
+
+    if (error) {
+      console.error("Error logging in:", error);
+      return;
+    }
+
+    const { data, error: fetchError } = await supabase.from("users").select("userType").eq("id", user.id).single();
```

Comment:
We do not have such column for users. However their `id` matches their GitHub ID. From there, you can fetch the user details and check they level inside the Organization, `member` or `collborator` etc.

Reply:
Please whats the column users are stored on in the supabase instance? Thanks

Reply:
The architecture of the project and how all the pieces communicate is a bit confusing to me. Please how can I get the schema correctly. Thanks

Reply:
The project architecture looks something similar to this
https://github.com/ubiquibot/.github/pull/11#issuecomment-2041143856

And the Supabase schema can be found at
https://github.com/ubiquity/ubiquibot/tree/development/supabase